Geolocation: strip whitespace when obtaining external IP

thejamescollins opened the pull request:
The "ip.appspot" service returns an IP address plus a  new line character in its response.

An example response from "ip.appspot" is:

`123.123.123.123\n`

This fix strips whitespace from the response before proceeding.

Other IP Lookup APIs don't seem to include a newline in the response, which means the trim() should be harmless in those cases.

thejamescollins committed:
Geolocation: strip whitespace when obtaining external IP

The "ip.appspot" service returns an IP address plus a  new line character in its response.

An example response from "ip.appspot" is:

```123.123.123.123\n```

This fix strip whitespace from the response before proceeding.

om4james commented:
Note: in order to replicate this bug, you need to be using a "local" install where the default IP address looks like a local IP (ie a virtual machine), which causes WooCommerce to fall back to determining the external IP.

It may also be necessary to retry multiple times to ensure that the "ip.appspot" endpoint is queried. In between retries I had to delete the `external_ip_address_***` transient to force WooCommerce to try another endpoint.

claudiosmweb commented:
@mikejolley maybe is better use `wc_clean()`.

thejamescollins committed:
Merge branch 'master' into geolocation-external-ip-fix

thejamescollins committed:
Use wc_clean to trim and sanitize external IP

As per suggestion from @claudiosmweb, wc_clean() is more full featured than trim().

See #7335
